### PR TITLE
Disallow`is_write` cards in dashboards alerts etc

### PR DESCRIPTION
**Pull request by camsaul**
Implements #22846

This PR throws a 400 if you attempt to add an `is_write` (QueryAction) Card to a Dashboard, make an Alert for it, share it publicly, or embed it.

### CODE REVIEW

File: test/metabase/api/card_test.clj [hunk 1]
```diff
@@ -384,6 +384,21 @@
...
+      (doseq [enable-embedding? [true false]]
+        (mt/with-temporary-setting-values [enable-embedding enable-embedding?]
+          (mt/with-model-cleanup [Card]
+            (is (schema= {:enable_embedding (s/eq false)
+                          s/Keyword         s/Any}
+                         (mt/user-http-request :crowberto :post 202 "card" {:name                   "My Card"
+                                                                            :display                :table
+                                                                            :dataset_query          (mt/mbql-query venues)
+                                                                            :visualization_settings {}
+                                                                            :enable_embedding       true})))))))))
```
Comment (camsaul) [inline test/metabase/api/card_test.clj:400]: This is not part of this change but I realized we weren't testing this so I added a test for it to make sure it was doing the right thing

**Review by escherize — APPROVED** [review]
